### PR TITLE
docs(csharp): add guides

### PR DESCRIPTION
**Pull request by morganleroi**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1812](https://algolia.atlassian.net/browse/DI-1812)

### Changes included:
- Add all C# examples for all guides.

[DI-1812]: https://algolia.atlassian.net/browse/DI-1812?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/clients/guides/filtering-your-search.mdx [hunk 1]
```diff
@@ -126,6 +126,19 @@ taskResponse, err := searchClient.WaitForTask(
...
+<TabItem value="csharp">
+
+```csharp
+using Algolia.Search.Clients;
+using Algolia.Search.Models.Search;
+
+var client = new SearchClient("<YOUR_APP_ID>", "<YOUR_API_KEY>");
+await client.SetSettingsAsync("YOUR_INDEX_NAME", new IndexSettings {
+  AttributesForFaceting = new List<string> { "actor", "filterOnly(category)", "searchable(publisher)" }
+    });
```
Comment (millotp) [inline website/docs/clients/guides/filtering-your-search.mdx:138]:
```suggestion
});
```

File: website/docs/clients/migration-guides/index.mdx [hunk 2]
```diff
@@ -101,6 +101,23 @@ client.search(
...
 ```
 
+</TabItem>
+<TabItem value="csharp">
+
+```csharp
+var client = new SearchClient("<YOUR_APP_ID>", "<YOUR_API_KEY>");
+
+client.Search<Hit>(new SearchMethodParams(new List<SearchQuery>
+  {
```
Comment (millotp) [inline website/docs/clients/migration-guides/index.mdx:111]: this indentation is a bit messed up here

File: website/docs/clients/migration-guides/index.mdx [hunk 3]
```diff
@@ -173,6 +190,16 @@ AbtestingClient client = new AbtestingClient("<YOUR_APP_ID>", "<YOUR_API_KEY>");
 ABTest abTest = client.getABTest(42);
 ```
 
+</TabItem>
+<TabItem value="csharp">
+
+  ```csharp
```
Comment (millotp) [inline website/docs/clients/migration-guides/index.mdx:196]: can you remove the spaces here and at the end pls ?

File: website/docs/clients/migration-guides/index.mdx [hunk 4]
```diff
@@ -615,6 +709,42 @@ for (SynonymHit synonym : synonyms) {
 ```
 
 </TabItem>
+
+<TabItem value="csharp" >
+
+  ```csharp
```
Comment (millotp) [inline website/docs/clients/migration-guides/index.mdx:715]: same here

File: website/docs/clients/migration-guides/csharp.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+### Usage
+
+To get started, first install the `Algolia.Search` client.
+
+You can get the last version of the client from [NuGet](https://www.nuget.org/packages/Algolia.Search/).
+
+If you are using the .NET CLI, you can install the package using the following command:
+
+```bash
+dotnet add package Algolia.Search --version [The version you want to install]
```
Comment (millotp) [inline website/docs/clients/migration-guides/csharp.md:14]: usually we have `<>` for required parameters

File: website/docs/clients/guides/wait-for-a-task-to-finish.mdx [hunk 6]
```diff
@@ -210,4 +210,28 @@ if err != nil {
...
+
+<TabItem value="csharp">
+
+```csharp
+using Algolia.Search.Clients;
+using Algolia.Search.Models.Search;
+using Algolia.Search.Utils;
+
+var response = await client.SaveObjectAsync("<YOUR_INDEX_NAME>",
+new { firstName = "john", lastname = "doe" });
```
Comment (millotp) [inline website/docs/clients/guides/wait-for-a-task-to-finish.mdx:222]:
```suggestion
  new { firstName = "john", lastname = "doe" });
```

File: website/docs/clients/guides/send-data-to-algolia.mdx [hunk 7]
```diff
@@ -297,4 +309,39 @@ if err != nil {
...
+<TabItem value="csharp">
+```csharp
+using System.Runtime.Serialization;
+using Algolia.Search.Clients;
+using Algolia.Search.Models.Search;
+using Action = Algolia.Search.Models.Search.Action;
+
+public class Actor : Hit
+{
+  [DataMember(Name = "name")]
```
Comment (millotp) [inline website/docs/clients/guides/send-data-to-algolia.mdx:322]: is this needed ?

File: website/docs/clients/guides/send-data-to-algolia.mdx [hunk 8]
```diff
@@ -79,6 +79,18 @@ searchClient, _ := search.NewClient("<APPLICATION_ID>", "<API_KEY>")
 ```
 
 </TabItem>
+
+<TabItem value="csharp">
+
+  ```csharp
```
Comment (millotp) [inline website/docs/clients/guides/send-data-to-algolia.mdx:85]: spaces

**Review by millotp — COMMENTED** [review]
the content is really good !
My only remark is about indentation, I didn't spot everything but it shows up in docusaurus, so it would be nice to have consistent indents

**Review by shortcuts — COMMENTED** [review]
nothing to add it looks pretty nice!

there seems to be some spacing issue, maybe copy all your examples in the playground and format them, the re-copy them in the doc?

File: website/docs/clients/guides/send-data-to-algolia.mdx (hunk 7 above)
Comment (morganleroi) [reply website/docs/clients/guides/send-data-to-algolia.mdx:322]: Yes. It's the way we tell the serializer that we want to serialize the field.

File: website/docs/clients/migration-guides/csharp.md (hunk 5 above)
Comment (morganleroi) [reply website/docs/clients/migration-guides/csharp.md:14]: I don't understand what you mean.

Comment (millotp) [reply website/docs/clients/migration-guides/csharp.md:14]: `<The version you want to install>`

Comment (morganleroi) [reply website/docs/clients/migration-guides/csharp.md:14]: oh okay understood

File: website/docs/clients/guides/send-data-to-algolia.mdx (hunk 7 above)
Comment (morganleroi) [reply website/docs/clients/guides/send-data-to-algolia.mdx:322]: Ok I have an issue. Dam.

File: website/docs/clients/guides/customized-client-usage.mdx [hunk 9]
```diff
@@ -485,5 +551,18 @@ searchClient, _ := search.NewClientWithConfig(search.Configuration{
 })
 ```
 
+</TabItem>
+
+<TabItem value="csharp">
+
+  > Create your own requester by creating a class that implements `IHttpRequester`, and use it by passing it to the client constructor.
```
Comment (shortcuts) [inline website/docs/clients/guides/customized-client-usage.mdx:558]:
```suggestion
> Create your own requester by creating a class that implements `IHttpRequester`, and use it by passing it to the client constructor.
```

**Review by shortcuts — COMMENTED** [review]
wonderful!

**Review by millotp — APPROVED** [review]

File: website/docs/clients/guides/customized-client-usage.mdx (hunk 9 above)
Comment (morganleroi) [reply website/docs/clients/guides/customized-client-usage.mdx]: Good catch !

**Review by millotp — APPROVED** [review]

**Pull request merged**